vault/identity_store_util.go

```diff
@@ -19,24 +20,57 @@ import (
 	"github.com/hashicorp/vault/logical"
 )
 
+var (
+	errDuplicateIdentityName = errors.New("duplicate identity name")
+)
+
 func (c *Core) loadIdentityStoreArtifacts(ctx context.Context) error {
-	var err error
 	if c.identityStore == nil {
 		c.logger.Warn("identity store is not setup, skipping loading")
 		return nil
 	}
 
-	err = c.identityStore.loadEntities(ctx)
-	if err != nil {
+	loadFunc := func(context.Context) error {
+		err := c.identityStore.loadEntities(ctx)
+		if err != nil {
+			return err
+		}
+		return c.identityStore.loadGroups(ctx)
+	}
+
+	// Load everything when memdb is set to operate on lower cased names
+	err := loadFunc(ctx)
+	switch {
+	case err == nil:
+		// If it succeeds, all is well
+		return nil
+	case err != nil && !errwrap.ContainsType(err, errDuplicateIdentityName):
+		// Return all errors except one
```

Comment:
What does this comment mean, are there multiple errors that we expect?

Reply:
It meant to return all the errors expect when error contains `errDuplicateIdentityName`.